test/functorch/test_control_flow.py

```diff
@@ -1314,6 +1396,220 @@ def mul(x: torch.Tensor, y: torch.Tensor):
             )
         self.assertEqual(cumsum1, cumsum_exp)
 
+    def test_scan_RNN(self):
+        dim = 1
+        device = torch.device("cpu")
+
+        rnn = torch.nn.RNN(
+            input_size=5,
+            hidden_size=7,
+        )
+        rnn = rnn.to(device=device)
+        x = torch.randn(1, 2, 5, device=device)
+        h = torch.randn(1, 2, 7, device=device)
+
+        new_state_dict = {
+            "weight_ih_l0": torch.ones_like(rnn.weight_ih_l0),
+            "bias_ih_l0": torch.ones_like(rnn.bias_ih_l0),
+            "weight_hh_l0": torch.ones_like(rnn.weight_hh_l0),
+            "bias_hh_l0": torch.ones_like(rnn.bias_hh_l0),
+        }
+        rnn.load_state_dict(new_state_dict)
+
+        def RNN(x: torch.Tensor, y: torch.Tensor):
+            W_ih = torch.ones((5, 7), device=device)
```

Comment:
We need to support closures/nn_modules in order for an actual RNN model to work. Please left a TODO for supporting closures.

Reply:
I have added a TODO for this in scan. This test was just the first attempt to realize an RNN and it somewhat works.